### PR TITLE
fix: Add missing github ping webhook event handler (#7555)

### LINKED ISSUE #7555: GitHub WebHook ping event: Webhook processing failed: event not defined to be parsed
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

**To Reproduce**

A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue.

```text
Request URL: <WebHook URL>
Request method: POST
Accept: */*
content-type: application/json
User-Agent: GitHub-Hookshot/04c124e
X-GitHub-Delivery: 33f01390-36cd-11ec-94f4-a0896a480161
X-GitHub-Event: ping
X-GitHub-Hook-ID: 325405283
X-GitHub-Hook-Installation-Target-ID: 421364558
X-GitHub-Hook-Installation-Target-Type: repository
X-Hub-Signature: sha1=f92aec6a8e42100292bab7d3a155d0dfd91c63b4
X-Hub-Signature-256: sha256=592c7264f830738bde2fe6eec8341b2e343183608dc0e44c84311959de459343
```

**Expected behavior**

`Response 200`

**Screenshots**

**Version**

```shell
Paste the output from `argocd version` here.
argocd: v2.1.3+d855831
  BuildDate: 2021-09-29T21:51:21Z
  GitCommit: d855831540e51d8a90b1006d2eb9f49ab1b088af
  GitTreeState: clean
  GoVersion: go1.16.5
  Compiler: gc
  Platform: linux/amd64
```

**Logs**

```
Paste any relevant application logs here.
```

### PR DESCRIPTION
**Pull request by jarviliam**
Signed-off-by: Liam Jarvis <jarviliam@gmail.com>

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Fixes #7555

### CODE REVIEW

File: applicationset/utils/testdata/github-ping-event.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
+{
```
Comment (crenshaw-dev) [inline applicationset/utils/testdata/github-ping-event.json:1]: Is this file used anywhere?

File: util/webhook/github-ping-event.json [hunk 2]
```diff
@@ -0,0 +1,142 @@
+{
```
Comment (crenshaw-dev) [inline util/webhook/github-ping-event.json:1]: Can you move this file to a `testdata` directory?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
A couple nitpicks. Thanks @jarviliam!

File: util/webhook/github-ping-event.json (hunk 2 above)
Comment (jarviliam) [reply util/webhook/github-ping-event.json]: Sure! I also moved all the other webhook event data with it and updated their tests 👍

File: applicationset/utils/testdata/github-ping-event.json (hunk 1 above)
Comment (jarviliam) [reply applicationset/utils/testdata/github-ping-event.json:1]: It's being used in the applicationset controller's tests:https://github.com/argoproj/argo-cd/blob/afd50954ef26bb6b235a93be8bb8141a97e27394/applicationset/utils/webhook_test.go#L56

It's the same test data as the one I added in argocd's webhook tests (argocd and app-controller each have their own respective webhook testdata it seems). Did you want the application-set controller's webhook tests to use the test data in argocd's webhook tests folder? Or do you want them to remain separate?

Comment (crenshaw-dev) [reply applicationset/utils/testdata/github-ping-event.json:1]: Nah, this works! I just didn't see the reference.

**Review by crenshaw-dev — APPROVED** [review]
lgtm, thanks!

**Pull request merged**